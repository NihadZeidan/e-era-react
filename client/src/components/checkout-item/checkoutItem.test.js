import { shallow } from "enzyme";
import CheckoutItem from "./checkoutItem.component";
import * as reactRedux from "react-redux";

const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");

beforeEach(() => {
  useSelectorMock.mockClear();
  useDispatchMock.mockClear();
});

it("Should render CheckoutItem Component", () => {
  const mockedDispatch = jest.fn();
  useDispatchMock.mockReturnValue(mockedDispatch);

  const mockCartItem = {
    name: "test name",
    price: 5,
    imageUrl: "test url",
    quantity: 1,
  };

  const wrapper = shallow(<CheckoutItem cartItem={mockCartItem} />);

  expect(wrapper).toMatchSnapshot();
});
