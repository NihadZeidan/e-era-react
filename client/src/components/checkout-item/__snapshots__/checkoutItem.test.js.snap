// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CheckoutItem Component 1`] = `ShallowWrapper {}`;
